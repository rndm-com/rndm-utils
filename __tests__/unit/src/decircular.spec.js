const decircular = require('../../../src/decircular');
const { describe, test } = global;

describe('src/decircular', () => {
  test('simple', () => {
    const input = { test: 'string' };
    const sut = decircular(input);
    expect(sut).toMatchSnapshot()
  });

  test('simple - spacer = 2', () => {
    const input = { test: 'string' };
    const sut = decircular(input, 2);
    expect(sut).toMatchSnapshot()
  });

  test('referenced', () => {

    const ref1 = { test: 'string1' };
    const ref2 = { test: 'string2' };

    ref1.ref2 = ref2;
    ref2.re1 = ref1;

    const sut = decircular(ref1);
    expect(sut).toMatchSnapshot()
  });

});
