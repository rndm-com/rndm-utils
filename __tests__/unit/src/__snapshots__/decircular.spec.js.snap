// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/decircular referenced 1`] = `
"{
  \\"test\\": \\"string1\\",
  \\"ref2\\": {
    \\"test\\": \\"string2\\"
  }
}"
`;

exports[`src/decircular simple - spacer = 2 1`] = `
"{
  \\"test\\": \\"string\\"
}"
`;

exports[`src/decircular simple 1`] = `
"{
  \\"test\\": \\"string\\"
}"
`;
